src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -828,3 +1009,30 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 	heap_close(pgDistColocation, NoLock);
 	relation_close(distributedRelation, NoLock);
 }
+
+
+/*
+ * SingleReplicatedTable checks whether all shards of a distributed table, do not have
+ * more than one replica. If even one shard has more than one replica, this function
+ * returns false, otherwise it returns true.
+ */
+static bool
+SingleReplicatedTable(Oid relationId)
```

Comment:
This seems a lot like the IsReplicated function in multi_router_executor. Maybe move that function into shardinterval_utils.c